src/cpp/shared/artifacts.cpp

```diff
@@ -15,437 +19,72 @@
-	"{Endless pouch of crystal}\n\nThe Endless Pouch of Crystal provides 1 unit of crystal/day.",
-	"{Spiked helm}\n\nThe Spiked Helm increases your attack and defense skills by 1 each.",
-	"{Spiked shield}\n\nThe Spiked Shield increases your attack and defense skills by 2 each.",
-	"{White pearl}\n\nThe White Pearl increases your spell power and knowledge by 1 each.",
-	"{Black pearl}\n\nThe Black Pearl increases your spell power and knowledge by 2 each.",
-	"{Magic Book}\n\nThe Magic Book enables you to cast spells.",
-	"{ERROR}\n\nArtifact 82.",
-	"{ERROR}\n\nArtifact 83.",
-	"{ERROR}\n\nArtifact 84.",
-	"{ERROR}\n\nArtifact 85.",
-	"{Spell Scroll}\n\nThis Spell Scroll gives your hero the ability to cast the '%s' spell.",
-	"{Arm of the Martyr}\n\nThe Arm of the Martyr increases your spell power by 3 but adds the undead morale penalty.",
-	"{Breastplate of Anduran}\n\nThe Breastplate increases your defense by 5.",
-	"{Broach of Shielding}\n\nThe Broach of Shielding provides 50% protection from Armageddon and Elemental Storm, but decreases spell power by 2.",
-	"{Battle Garb}\n\nThe Battle Garb of Anduran combines the powers of the three Anduran artifacts.  It provides maximum luck and morale for your troops and gives you the Town Portal spell.",
-	"{Crystal Ball}\n\nThe Crystal Ball lets you get more specific information about monsters, enemy heroes, and castles nearby the hero who holds it.",
-	"{Heart of Fire}\n\nThe Heart of Fire provides 50% protection from fire, but doubles the damage taken from cold.",
-	"{Heart of Ice}\n\nThe Heart of Ice provides 50% protection from cold, but doubles the damage taken from fire.",
-	"{Helmet of Anduran}\n\nThe Helmet increases your spell power by 5.",
-	"{Holy Hammer}\n\nThe Holy Hammer increases your attack skill by 5.",
-	"{Legendary Scepter}\n\nThe Legendary Scepter adds 2 points to all attributes.",
-	"{Masthead}\n\nThe Masthead boosts your luck and morale by 1 each in sea combat.",
-	"{Sphere of Negation}\n\nThe Sphere of Negation disables all spell casting, for both sides, in combat.",
-	"{Staff of Wizardry}\n\nThe Staff of Wizardry boosts your spell power by 5.",
-	"{Sword Breaker}\n\nThe Sword Breaker increases your defense by 4 and attack by 1.",
-	"{Sword of Anduran}\n\nThe Sword increases your attack skill by 5.",
-	"{Spade of Necromancy}\n\nThe Spade gives you increased necromancy skill.",
-	"{Pandora's Box}\n\nPandora's Box gives your hero's army a random stack of level 1 creatures for the duration of each battle.",
-    "{Iron Fist}\n\nThe Iron Fist is a snazzy accessory to go with your helmet and boots.",
-};
+  int TotalArtifacts(const artifacts::artifact_sequence &artifactList) {
+    int numArtifacts = 0;
+    for (const auto &art : artifactList) {
+      const int i = art.id();
+      if (i > numArtifacts && i < NUM_SUPPORTED_ARTIFACTS) {
+        numArtifacts = i;
+      }
+    }
+    return numArtifacts + 1;
```

Comment:
This will return 1 even if there are 0 artifacts. It also has problems if we make artifacts that skip IDs; perhaps this is misnamed.

Reply:
Naming things is hard.  I'll make it clearer.